src/java_tools/junitrunner/java/com/google/testing/coverage/JacocoCoverageRunner.java

```diff
@@ -345,7 +346,30 @@ private static String getUniquePath(String pathTemplate, String suffix) throws I
     return convertedMetadataFiles.build();
   }
 
-  private static URL[] getUrls(ClassLoader classLoader) {
+  private static URL[] getUrls(ClassLoader classLoader, boolean wasWrappedJar) {
+    URL[] urls = getClassLoaderUrls(classLoader);
+    // If the classpath was too long then a temporary top-level jar is created containing nothing but a manifest with
+    // the original classpath. For the moment, we just assume it's made up of URLs to other jars and extract them.
+    if (urls != null) {
+      if (wasWrappedJar && urls.length == 1) {
+        try {
+          String jarClassPath =
+              new JarInputStream(urls[0].openStream()).getManifest().getMainAttributes().getValue("Class-Path");
+          String[] urlStrings = jarClassPath.split(" ");
+          URL[] newUrls = new URL[urlStrings.length];
+          for (int i = 0; i < urlStrings.length; i++) {
+            newUrls[i] = new URL(urlStrings[i]);
+          }
+          return newUrls;
+        } catch (Exception e) {
+         return null;
```

Comment:
There's not much error handling in the surrounding code, but there is a printStackTrace, which you might want to have.

Reply:
The error handling in this entire region can be described succinctly with the statement "Hope nothing goes wrong".